aten/src/ATen/native/SortingUtils.h

```diff
@@ -4,6 +4,8 @@
 #include <ATen/native/Resize.h>
 #include <c10/util/irange.h>
 
+#include <ATen/ops/empty.h>
```

Comment:
Do you have an automatic way to populate these? Do you think we want to (can?) have one so that we can keep these minimal over time?

Reply:
@albanD I remember Ed mentioning some problems with his ufunc codegen PR around internal buck targets not being able to handle dynamically codegen'd headers. Did that not turn out to be a problem here with the per-op headers?

Reply:
I just imported this stack. Will see how it goes. Hopefully not...
But potentially, we can gate it on cmake version / internal and the internal incremental build will just be not as good.

Reply:
The headers required special handling to work with `buck`/`bazel`. See the build system changes here:
https://github.com/pytorch/pytorch/pull/68247/files#diff-ff6cf959d408c4356a73d786776adac2a9a39a90ed1689f392e8ec23984e0764

`genrule` ("general rule") requires the list of output files to be statically known, but with  `rule` you can create your own rules that are more flexible. Here the only difference is I specify an entire directory as an output. This has shown to work this with `bazel` locally and in CI, hopefully `buck` doesn't diverge too much here.

Reply:
>Do you have an automatic way to populate these? Do you think we want to (can?) have one so that we can keep these minimal over time?

That would be really useful. I know there is [include-what-you-use](https://include-what-you-use.org/) which does this for headers generally. Not sure if you're able to limit it to run on specific kinds of includes.